core/adapter/reslice.h

```diff
@@ -183,10 +186,37 @@ namespace MR
         default_type from[3], inc[3];
         default_type norm;
         const transform_type transform_, direct_transform;
+
+        void normalise (default_type&) const;
```

Comment:
You could rewrite this as a small external helper function, templated on your `value_type`:
~~~C++
namespace { // in anonymous namespace to avoid any chance of external conflict
  template <typename X> inline void normalise (default_type&) const;
}
~~~

and invoke the explicit specialisation you're after at line 163:
~~~C++
normalise<value_type> (result);
~~~
That should simplify your SFINAE somewhat...? Also, you could have a default implementation, a specialisation for `bool`, and a SFINAE specialisation for all integer types - I think C++ would allow for that? The only catch would be to make sure that the SFINAE specialisation doesn't also catch `bool` - but you already check that anyway...